src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -31,11 +31,12 @@ internal class FunctionExecutor : IFunctionExecutor
         private readonly IAsyncCollector<FunctionInstanceLogEntry> _functionEventCollector;
         private readonly ILogger _logger;
         private readonly ILogger _resultsLogger;
+        private readonly JobHost _host;
 
         private HostOutputMessage _hostOutputMessage;
 
         public FunctionExecutor(IFunctionInstanceLogger functionInstanceLogger, IFunctionOutputLogger functionOutputLogger,
-                IWebJobsExceptionHandler exceptionHandler, TraceWriter trace, IAsyncCollector<FunctionInstanceLogEntry> functionEventCollector = null,
+                IWebJobsExceptionHandler exceptionHandler, TraceWriter trace, JobHost host = null, IAsyncCollector<FunctionInstanceLogEntry> functionEventCollector = null,
```

Comment:
As discussed, we should have a new IJobMethodInvoker interface that is passed in here. It must be specified, so you shouldn't have "= null" here.